code/obj/lattice.dm

```diff
@@ -35,15 +35,18 @@
 			else
 
 	proc/replace_with_catwalk(var/obj/item/rods/rods)
-		var/turf/simulated/floor/airless/plating/catwalk/auto/T = get_turf(src.loc)
-		T.ReplaceWith(/turf/simulated/floor/airless/plating/catwalk/auto, keep_old_material = 0, handle_dir = 1)
-		if(istype(T)) // ReplaceWith can fail if unsim turf etc
+		var/turf/simulated/floor/T = get_turf(src.loc)
+		var/replacement = /turf/simulated/floor/airless/plating/catwalk/auto
+		if (!istype(T, /turf/unsimulated/floor))
+			if (istype(T, /turf/space/fluid) || istype(T, /turf/simulated/floor))
+				replacement = /turf/simulated/floor/airless/plating
```

Comment:
Why would it replace with airless plating? Surely it could just keep the ocean tile?

Reply:
plating seemed to make sense, do you think floors should also not replace turf?